src/plugins/intel_gna/src/layers/gna_convolution_layer.cpp

```diff
@@ -17,12 +17,22 @@
 namespace GNAPluginNS {
 namespace GNAConvolutionLayer {
 
+bool should_transpose_h_w(const uint32_t in_height,
+                          const uint32_t kernel_height,
+                          const uint32_t in_channels,
+                          const uint32_t stride_height) {
+    return in_height == kernel_height && in_channels == 1 && stride_height == 1;
+}
+
 bool isMappableFrom2DTo1D(const uint32_t inHeight, const uint32_t inWidth, const uint32_t in_channels,
                           const uint32_t kernelHeight, const uint32_t kernelWidth,
                           const uint32_t strideHeight, const uint32_t strideWidth) {
-    return ((inHeight > 1 && inWidth > 1) &&
-            ((inWidth == kernelWidth && strideWidth == 1) ||
-             (inHeight == kernelHeight && strideHeight == 1 && in_channels == 1)));
+    if (inHeight <= 1 || inWidth <= 1) {
+        // Mapping from 2D-> 1D is only needed if the input is non-trivial 2D
```

Comment:
What is a trivial 2D? I guess we should rather use rank term here.

Reply:
Rank is not ambiguous in the OV domain. It's used in the Convolution layer documentation: "1D convolution (input tensors rank 3) means that there is only one spatial axis X". But 1D is OK IMO as well. I thought you didn't want to use 1D, because you put "trivial 2D".